src/extension/app/store/app.js

```diff
@@ -532,7 +532,22 @@ export class AppStore {
    */
   getContentSourceLabel() {
     const { preview } = this.status;
-    return preview?.sourceLocation.includes('onedrive:') ? 'SharePoint' : 'Google Drive';
+    const { sourceLocation } = preview;
+
+    if (sourceLocation) {
+      return sourceLocation.includes('onedrive:')
+        ? 'SharePoint'
+        : sourceLocation.includes('gdrive:')
+          ? 'Google Drive'
+          : 'BYOM';
+    }
+
+    const { mountpoint } = this.siteStore;
+    return mountpoint.includes('.sharepoint.com')
+      ? 'SharePoint'
+      : mountpoint.includes('.google.com')
+        ? 'Google Drive'
+        : 'BYOM';
```

Comment:
I think here it's safer to check for `.google.com` first, and then try to differentiate between SharePoint and BYOM for the rest. Since custom SharePoint URLs don't have to have a `host` ending in `.sharepoint.com`, it might be wise to add an OR check for something like `/Shared%20Documents/sites/` in the `pathname` before falling back to BYOM.